external/storm-cassandra/src/test/java/org/apache/storm/cassandra/trident/TridentTopologyTest.java

```diff
@@ -0,0 +1,125 @@
+
+import static org.apache.storm.cassandra.DynamicStatementBuilder.boundQuery;
+import static org.apache.storm.cassandra.DynamicStatementBuilder.field;
+import static org.apache.storm.cassandra.DynamicStatementBuilder.with;
+
+/**
+ *
+ */
+public class TridentTopologyTest extends BaseTopologyTest {
+
+    @Test
+    public void testTridentTopology() throws Exception {
+
+        Session session = cassandraCQLUnit.session;
+        String[] stationIds = {"station-1", "station-2", "station-3"};
+        for (int i = 1; i < 4; i++) {
+            ResultSet resultSet = session.execute("INSERT INTO weather.station(id, name) VALUES(?, ?)", stationIds[i-1],
+                    "Foo-Station-" + new Random().nextInt());
+        }
+
+        ResultSet rows = cassandraCQLUnit.session.execute("SELECT * FROM weather.station");
+        for (Row row : rows) {
+            System.out.println("####### row = " + row);
+        }
+
+        WeatherBatchSpout weatherBatchSpout =
+                new WeatherBatchSpout(new Fields("weather_station_id", "temperature", "event_time"), 3,
+                        stationIds);
+
+        TridentTopology topology = new TridentTopology();
+        Stream stream = topology.newStream("cassandra-trident-stream", weatherBatchSpout);
+
+        CassandraStateFactory insertValuesStateFactory = getInsertTemperatureStateFactory();
+
+        CassandraStateFactory selectWeatherStationStateFactory = getSelectWeatherStationStateFactory();
+
+        TridentState selectState = topology.newStaticState(selectWeatherStationStateFactory);
+        stream = stream.stateQuery(selectState, new Fields("weather_station_id"), new CassandraQuery(), new Fields("name"));
+        stream = stream.each(new Fields("name"), new PrintFunction(), new Fields("name_x"));
+
+        stream.partitionPersist(insertValuesStateFactory, new Fields("weather_station_id", "name", "event_time", "temperature"), new CassandraStateUpdater(), new Fields());
+
+        StormTopology stormTopology = topology.build();
+        LocalCluster cluster = new LocalCluster();
+        cluster.submitTopology("wordCounter", getConfig(), stormTopology);
+        Thread.sleep(30 * 1000);
+
+        rows = cassandraCQLUnit.session.execute("SELECT * FROM weather.temperature");
+        Assert.assertTrue(rows.iterator().hasNext()); // basic sanity check
+
+        cluster.killTopology("wordCounter");
+        cluster.shutdown();
+    }
+
+    public static class PrintFunction extends BaseFunction {
+
+        @Override
+        public void execute(TridentTuple tuple, TridentCollector collector) {
+            System.out.println("####### tuple = " + tuple.getValues());
+            collector.emit(tuple.getValues());
+        }
+    }
+
+    private CassandraStateFactory getInsertTemperatureStateFactory() {
+        CassandraState.Options options = new CassandraState.Options(new CassandraContext());
+        CQLStatementTupleMapper insertTemperatureValues = boundQuery(
```

Comment:
Cleaning up these test cases is part of another JIRA STORM-1348. The above changes are done as part of refactoring the cassandra table script.